boltdb_batch.go

```diff
@@ -12,19 +12,37 @@ type boltDBBatch struct {
 
 var _ Batch = (*boltDBBatch)(nil)
 
+func newBoltDBBatch(db *BoltDB) *boltDBBatch {
+	return &boltDBBatch{
+		db:  db,
+		ops: make([]operation, 0, 1),
+	}
+}
+
 // Set implements Batch.
-func (b *boltDBBatch) Set(key, value []byte) {
+func (b *boltDBBatch) Set(key, value []byte) error {
+	if b.ops == nil {
+		return ErrBatchClosed
```

Comment:
what do u think about panicking here?

Reply:
Yeah, I was thinking about that too. Not really sure - I don't like all of the panicing we do, but in this case it might be appropriate: we wouldn't need to change the interface, and callers wouldn't have to check for errors all the time.

The standard library does both: writing to a closed channel panics, but writing to a closed `io.Writer` returns an error.

@tessr @marbar3778 What do you guys think?

Reply:
Discussed this with @marbar3778 on Slack. Breaking the interface here doesn't actually buy us anything useful, since none of the Tendermint/SDK/IAVL uses of `Batch` are affected by this. I suggest we panic for now, and consider changing the interface when we have to break it for an actually useful reason.